### PR TITLE
fix(tags): improve tag manager

### PR DESCRIPTION
**Pull request by Yelinz**
improve tag service by moving tag manager parts out
improved error handling in tag service

requires backend change  https://github.com/projectcaluma/alexandria/pull/387

### CODE REVIEW

File: addon/services/tags.js [hunk 1]
```diff
@@ -45,14 +41,19 @@ export default class TagsService extends Service {
...
-  }
-
-  @task *fetchSearchTags() {
+  @task *fetchSearchTags(search) {
+    let withDocumentsInCategories;
+    if (this.category) {
+      const category =
+        this.store.peekRecord("category", this.category) ??
+        (yield this.store.findRecord("category", this.category));
+      const allCategories = [category, ...(yield category.children)];
```
Comment (czosel) [inline addon/services/tags.js:50]: not needed anymore when backend implementation covers this

File: addon/services/tags.js [hunk 2]
```diff
@@ -45,14 +41,19 @@ export default class TagsService extends Service {
       });
   }
 
-  @task *fetchAllTags() {
-    return yield this.store.findAll("tag");
-  }
-
-  @task *fetchSearchTags() {
+  @task *fetchSearchTags(search) {
```
Comment (czosel) [inline addon/services/tags.js:44]:
```suggestion
  @task *fetchSearchTags() {
```

File: addon/services/tags.js [hunk 3]
```diff
@@ -45,14 +41,19 @@ export default class TagsService extends Service {
...
+      const category =
+        this.store.peekRecord("category", this.category) ??
+        (yield this.store.findRecord("category", this.category));
+      const allCategories = [category, ...(yield category.children)];
+      withDocumentsInCategories = allCategories.map((c) => c.id).join(",");
+    }
     return yield this.store.query("tag", {
       filter: {
-        withDocumentsInCategory: this.category,
+        search,
```
Comment (czosel) [inline addon/services/tags.js:55]: not needed

File: addon/services/tags.js [hunk 4]
```diff
@@ -110,11 +117,17 @@ export default class TagsService extends Service {
    */
   @action async remove(document, tag) {
     if (typeof tag === "string") {
-      tag = this.allTags.find((t) => t.name === tag);
+      tag = this.store.peekRecord("tag", tag.id);
+      //tag = this.allTags.find((t) => t.name === tag);
```
Comment (czosel) [inline addon/services/tags.js:121]: drop the comment

Comment (Yelinz) [reply addon/services/tags.js]: All cleaned up

**Review by czosel — DISMISSED** [review]
As discussed: let's add some documentation for the new filter property, otherwise LGTM :+1:

**Pull request merged**